### PR TITLE
Fixed bug where we tried to read missing custom regex list

### PR DESCRIPTION
**Pull request by AndreasArvidsson**
## Checklist

- [/] I have added [tests](https://www.cursorless.org/docs/contributing/test-case-recorder/)
- [/] I have updated the [docs](https://github.com/cursorless-dev/cursorless/tree/main/docs) and [cheatsheet](https://github.com/cursorless-dev/cursorless/tree/main/cursorless-talon/src/cheatsheet)
- [/] I have not broken the cheatsheet

### CODE REVIEW

File: cursorless-talon/src/get_grapheme_spoken_form_entries.py [hunk 1]
```diff
@@ -37,6 +27,16 @@ def get_grapheme_spoken_form_entries(
 
 
 def get_graphemes_talon_list() -> dict[str, str]:
+    if grapheme_capture_name not in registry.captures:
```
Comment (AndreasArvidsson) [inline cursorless-talon/src/get_grapheme_spoken_form_entries.py:30]: After the merge of literal mark this if statement ended up in the wrong place. It should of course be here where we actually read it from the registry.

File: cursorless-talon/src/spoken_forms.py [hunk 2]
```diff
@@ -125,6 +125,7 @@ def handle_new_values(csv_name: str, values: list[SpokenFormEntry]):
         if initialized:
             # On first run, we just do one update at the end, so we suppress
             # writing until we get there
+            init_scope_spoken_forms(graphemes_talon_list)
```
Comment (AndreasArvidsson) [inline cursorless-talon/src/spoken_forms.py:128]: Without this we don't regenerate the flattened list whenever a csv file changes

File: cursorless-talon/src/spoken_scope_forms.py [hunk 3]
```diff
@@ -4,25 +4,20 @@
...
+def create_flattened_talon_list(ctx: Context, graphemes_talon_list: dict[str, str]):
     lists_to_merge = {
         "cursorless_scope_type": "simple",
         "cursorless_surrounding_pair_scope_type": "surroundingPair",
         "cursorless_selectable_only_paired_delimiter": "surroundingPair",
         "cursorless_wrapper_selectable_paired_delimiter": "surroundingPair",
     }
-    if include_custom_regex:
+    # If the user have no custom regex scope type, then that list is missing from the context
+    if "user.cursorless_custom_regex_scope_type" in ctx.lists.keys():  # noqa: SIM118
```
Comment (AndreasArvidsson) [inline cursorless-talon/src/spoken_scope_forms.py:20]: This is really weird. ruff wants me to remove `.keys()` but if I do this condition is always false. I wonder if this has anything to do with how Talon implements the lists/dict?

Comment (pokey) [reply cursorless-talon/src/spoken_scope_forms.py:20]: yeah SIM118 is fairly aggressive imo. Fine with a noqa but if it keeps biting us we might consider just shutting it off

**Review by pokey — APPROVED** [review]
it's a bit surprising that the list just doesn't get created at all if it's empty, but best to get this fix in and can re-evaluate that after

**Comment by AndreasArvidsson** [comment]
> it's a bit surprising that the list just doesn't get created at all if it's empty, but best to get this fix in and can re-evaluate that after

Yeah that was a bit surprising. I think we should have a real look over the entire csv handling and usage of list when you are back.

File: cursorless-talon/src/spoken_scope_forms.py (hunk 3 above)
Comment (AndreasArvidsson) [reply cursorless-talon/src/spoken_scope_forms.py:20]: I think it should be equivalent generally in python and this is something Talon is doing

**Pull request merged**